KlarnaOnDemand/KODOriginProof.m

```diff
@@ -3,12 +3,37 @@
 #import <Foundation/NSJSONSerialization.h>
 #import "NSData+Base64.h"
 
+@interface KODOriginProof ()
+
+
+@property(nonatomic, assign) int amount;
+@property(strong, nonatomic) NSString *currency;
+@property(strong, nonatomic) NSString *userToken;
+
+@end
+
 @implementation KODOriginProof
 
-+ (NSString *)generateWithAmount:(int)amount currency:(NSString *)currency userToken:(NSString *)userToken {
-  NSData *data = [self jsonDataWithDictionary:@{@"amount": [NSNumber numberWithInt:amount],
-                                                @"currency": currency,
-                                                @"user_token": userToken,
+
+- (id)initWithAmount:(int)amount currency:(NSString *)currency userToken:(NSString *)userToken {
+  self = [super init];
+  if (self) {
+    self.amount = amount;
+    self.currency = currency;
+    self.userToken = userToken;
+  }
+  return self;
+}
+
+- (id)init {
+  NSAssert(NO, @"Initialize with -initWithAmount:currency:userToken");
+  return nil;
+}
+
+- (NSString *)description {
+  NSData *data = [self jsonDataWithDictionary:@{@"amount": [NSNumber numberWithInt: self.amount],
```

Comment:
I'd move the actual origin proof generation to the constructor. Store the result and return it when calling `description`.

Also, maybe align these key/value pairs somehow?